File: fix_imports.py
import os
import re

def process_file(filepath):
    with open(filepath, 'r', encoding='utf8') as f:
        content = f.read()

    original = content
    has_backend = 'BACKEND_URL' in content
    has_api = 'API_BASE_URL' in content

    if not has_backend and not has_api:
        return

    missing = []
    
    if has_backend:
        if not re.search(r'import\s+\{.*BACKEND_URL.*\}\s+from\s+[\'\"]@/lib/api[\'\"]', content):
            missing.append('BACKEND_URL')
    
    if has_api:
        if not re.search(r'import\s+\{.*API_BASE_URL.*\}\s+from\s+[\'\"]@/lib/api[\'\"]', content):
            missing.append('API_BASE_URL')

    if missing:
        print(f'Fixing {filepath}: {missing}')
        if 'import { api } from "@/lib/api";' in content:
            content = content.replace('import { api } from "@/lib/api";', f'import {{ api, {", ".join(missing)} }} from "@/lib/api";')
        elif 'import { api' in content:
            # simple fallback
            content = f'import {{ {", ".join(missing)} }} from "@/lib/api";\n' + content
        else:
            content = f'import {{ {", ".join(missing)} }} from "@/lib/api";\n' + content

        with open(filepath, 'w', encoding='utf8') as f:
            f.write(content)

for root, dirs, files in os.walk('app'):
    for file in files:
        if file.endswith('.js') or file.endswith('.jsx'):
            process_file(os.path.join(root, file))

for root, dirs, files in os.walk('components'):
    for file in files:
        if file.endswith('.js') or file.endswith('.jsx'):
            process_file(os.path.join(root, file))

File: fix_imports.js
const fs = require('fs');
const path = require('path');

function processDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            processDir(fullPath);
        } else if (fullPath.endsWith('.js') || fullPath.endsWith('.jsx')) {
            processFile(fullPath);
        }
    }
}

function processFile(filePath) {
    let content = fs.readFileSync(filePath, 'utf8');
    let original = content;

    const hasBackendUrl = content.includes('BACKEND_URL');
    const hasApiBaseUrl = content.includes('API_BASE_URL');
    
    if (hasBackendUrl || hasApiBaseUrl) {
        let missingImports = [];
        if (hasBackendUrl && !content.includes(' BACKEND_URL') && !content.includes('{ BACKEND_URL') && !content.includes('BACKEND_URL,') && !content.includes('BACKEND_URL }') && !content.includes('{BACKEND_URL}')) {
             missingImports.push('BACKEND_URL');
        }
        if (hasApiBaseUrl && !content.includes(' API_BASE_URL') && !content.includes('{ API_BASE_URL') && !content.includes('API_BASE_URL,') && !content.includes('API_BASE_URL }') && !content.includes('{API_BASE_URL}')) {
             missingImports.push('API_BASE_URL');
        }
        
        if (missingImports.length > 0) {
            console.log('Missing in ' + filePath + ': ' + missingImports.join(', '));
            if (content.includes('import { api } from "@/lib/api";')) {
                content = content.replace('import { api } from "@/lib/api";', 'import { api, ' + missingImports.join(', ') + ' } from "@/lib/api";');
            } else if (content.includes('import { api, ') || content.includes(', api }')) {
                content = 'import { ' + missingImports.join(', ') + ' } from "@/lib/api";\n' + content;
            } else {
                content = 'import { ' + missingImports.join(', ') + ' } from "@/lib/api";\n' + content;
            }
            fs.writeFileSync(filePath, content);
        }
    }
}

const rootDir = __dirname;
processDir(path.join(rootDir, 'app'));
processDir(path.join(rootDir, 'components'));
